mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -105,7 +125,11 @@ class StringIndexer @Since("1.4.0") (
 
 @Since("1.6.0")
 object StringIndexer extends DefaultParamsReadable[StringIndexer] {
-
+  private[feature] val SKIP_UNSEEN_LABEL: String = "skip"
+  private[feature] val ERROR_UNSEEN_LABEL: String = "error"
+  private[feature] val KEEP_UNSEEN_LABEL: String = "keep"
```

Comment:
At some point, let's do that, but not yet.  I like keeping things private at first in case we find mistakes after release and need to change things.